src/components/Dataset/MoveToDatasetButton.js

```diff
@@ -1,74 +1,69 @@
-import { useState } from 'react'
 import { useRouter } from 'next/router'
 import { useDatasetManager } from 'hooks/useDatasetManager'
 import { useModal } from 'hooks/useModal'
 import formatNumbers from 'helpers/formatNumbers'
-import { Button } from 'components/shared/Button'
+import { Button as SharedButton } from 'components/shared/Button'
 import { Modal } from 'components/shared/Modal'
 import { MoveToDatasetModal } from './MoveToDatasetModal'
 
+const Button = ({ onClick }) => (
+  <SharedButton
+    label="Move to Dataset"
+    secondary
+    responsive
+    onClick={onClick}
+  />
+)
+
 export const MoveToDatasetButton = ({ dataset }) => {
   const { push } = useRouter()
+  const { openModal, closeModal } = useModal()
+  const modalId = `move-to-dataset-${dataset.id}`
   const {
     dataset: myDataset,
     addSamples,
-    getTotalSamples
+    getTotalSamples,
+    replaceSamples
   } = useDatasetManager()
-  const { openModal, closeModal } = useModal()
-  const id = `move-to-dataset-${dataset.id}`
-  const radioOptions = [
-    { label: 'Append samples to My Dataset', value: 'append' },
-    { label: 'Replace samples in My Dataset', value: 'replace' }
-  ]
-  const defaultValue = radioOptions[0].value
-  const [value, setValue] = useState(defaultValue)
-  const newTotalSamples = getTotalSamples(dataset.data)
-  const totalSamples = getTotalSamples(myDataset?.data)
-  const pathname = '/download'
+  const myDatasetTotalSamples = getTotalSamples(myDataset?.data)
+
+  const getRedirect = (prefix) => {
+    push({
+      pathname: '/download',
+      query: {
+        message: `${prefix} ${formatNumbers(
+          getTotalSamples(dataset.data) // display the shared dataset total samples
+        )} samples to My Dataset`,
+        status: 'success'
+      }
+    })
+  }
 
-  const handleMoveToDataset = async () => {
-    if (totalSamples > 0) {
-      openModal(id)
-    } else {
-      await addSamples(dataset.data)
-      push(
-        {
-          pathname,
-          query: {
-            message: `Appended ${formatNumbers(
-              newTotalSamples
-            )} samples to My Dataset`,
-            status: 'success'
-          }
-        },
-        pathname
-      )
-    }
+  const handleAppend = async () => {
+    await addSamples(dataset.data)
+    getRedirect('Appended')
+  }
+
+  const handleReplace = async () => {
+    await replaceSamples(dataset.data)
+    getRedirect('Moved')
+  }
```

Comment:
```suggestion
const handlers = {
  append: handleAppend,
  replace: async () => {
    await replaceSamples(dataset.data)
    downloadRedirectWithPrefix('Moved')
  }
}

const handleSelectAction = (action) => {
   await handlers[action]()
   // analytics
   // close modal
}
```

Reply:
I've defined the `handlers` map for the actions and moved the `action` state to the parent component  `MoveToDatasetButton`.

Additionally, I've used  the `<Form>` element in `MoveToDatasetModal` to handle submit and reset events, and updated the props to include `onChange`, `onReset`, and `onSubmit` for better clarity.  

This PR is ready for another look. Thank you, David!